networkit/cpp/correlation/test/AssortativityGTest.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * AssortativityGTest.cpp
+ *
+ *  Created on: 04.03.2022
+ *      Author: Lucas Petersen
+ */
+
+#include <gtest/gtest.h>
+
+#include <networkit/auxiliary/Log.hpp>
+#include <networkit/correlation/Assortativity.hpp>
+#include <networkit/io/METISGraphReader.hpp>
+
+namespace NetworKit {
+
+class AssortativityGTest : public testing::Test {};
+
+TEST_F(AssortativityGTest, testAssortivityPartitionSmallGraph) {
+    /* Graph:
+       0   3   7
+        \ /|\ /
+         2 5 6
+        / \|/ \
+       1   4   8
+    */
+    Graph G(9);
+
+    G.addEdge(0, 2);
+    G.addEdge(1, 2);
+    G.addEdge(2, 3);
+    G.addEdge(2, 4);
+    G.addEdge(3, 5);
+    G.addEdge(3, 6);
+    G.addEdge(4, 5);
+    G.addEdge(4, 6);
+    G.addEdge(6, 7);
+    G.addEdge(6, 8);
+
+    /*
+    Since we have 2 edges between each partition(except partion 0 and 2) and 2 edges within each.
+    This gives us the matrix(normalized by m):
+    --------
+    |0.2 0.2 0
+    |0.2 0.2 0.2
+    |0   0.2 0.2
+
+    The diagSum is then 0.6 and the ab coeff sum is 0.68
```

Comment:
```suggestion
    The diagSum is thus 0.6 and the ab coeff sum is 0.68
```
What exactly is the "ab coeff"?